### DISCUSSION TITLE
How should one generate datasets for MSA pretraining?

### DISCUSSION THREAD

**Discussion by leasunhy** [facebookresearch/esm#68] [Q&A] [answered]
Congrats for the awesome performance of MSA Transformer!

Recently I'm interested in reproducing your results by training an MSA Transformer from scratch, but I am stuck in the very first step: data preparation...

Could you show some hints on the following questions? Thank you! 😄 
* The UniRef50 dataset contains ~30M cluster representatives, but the number of data reported in the MSA Transformer paper is 26M. Why is the 4M difference?
* Did you split UniRef50 into two sets (train, held out) or three sets (train, dev, held out)? On which set did you tune the hyper-parameters of the model?

**Comment by tomsercu** [comment, marked as answer]
Hi, happy to help dig up the right pointers!
[This section of the repo README](https://github.com/facebookresearch/esm#pre-training-dataset-split--) provides the split which is also used for MSA Transformer; it splits the 2018 uniref50 into 3.016 M validation clusters and 27.145 M train clusters.
HHblits MSA generation with the settings described in the paper failed for some of those 27M, yielding the 26M MSAs used for training.

**Reply by leasunhy** [reply]
Thanks for the info! 😆
